### ISSUE TITLE
[FR] Support for Linux Host on ARM

### ISSUE THREAD

**Issue by rbebb** [android/ndk#1752]
### Description

I'm looking to use the NDK in an ARM Linux Docker container running on an Apple Silicon computer. However, it would be great to have support for any Linux Host on ARM.

**Comment by DanAlbert** [comment]
Some prior discussion: https://github.com/android/ndk/discussions/1692

Tl;Dr, no plans.

**Comment by rbebb** [comment]
@DanAlbert I noticed that this ticket is still open. Is it open because ARM support is being considered?

**Comment by DanAlbert** [comment]
No change since my last comment. It's open because it's a reasonable request that has yet to become a priority.

**Comment by Rodrigo-Vargas** [comment]
Hey @DanAlbert , any updates on this since your last comment?

**Comment by DanAlbert** [comment]
No. We will post an update when there's an update. We're well aware that people want this, but the infrastructure we need to make it happen just doesn't exist.

**Comment by HomuHomu833** [comment]
Hmm this is my first time seeing this issue, but let’s get things sorted out.

Unofficially, you can use [my project Android NDK Custom](https://github.com/HomuHomu833/android-ndk-custom) which builds almost everything (except `simpleperf`) for a variety of unusual architectures, including aarch64.

TL;DR of the concept: It uses [Zig](https://codeberg.org/ziglang/zig)'s LLVM Backend and [musl libc](https://musl.libc.org/) to build binaries and replaces the original ones.

Hopefully, Google releases official aarch64 NDK builds soon.

**Comment by finagolfin** [comment]
I see [a linux AArch64 toolchain being prepared](https://android.googlesource.com/platform/prebuilts/clang/host/linux-arm64/+/refs/heads/mirror-goog-main-prebuilts/clang-r596125/bin/): will that be in the next NDK 30 beta or some subsequent NDK? Looking forward to using it, as [AArch64 machines proliferate](https://www.electronicsweekly.com/news/business/arm-taking-25-of-server-cpu-market-2025-09/), currently using the macOS arm64 build too without issue.

**Comment by PawarisMopha** [comment]
> ### Description
> I'm looking to use the NDK in an ARM Linux Docker container running on an Apple Silicon computer. However, it would be great to have support for any Linux Host on ARM.
> 
> Components that need Linux Arm support:
> 
> * [x]  ndk-build[ ]  LLVM toolchain[ ]  CMake[ ]  Ninja[ ]  Make[ ]  Python[ ]  yasm (probably switch to nasm)[ ]  shader-tools[ ]  simpleperf

**Comment by Sharjeel-Khan** [comment]
We've brought up a linux/arm64 NDK and removed some tools in r31 in the process. Stay tuned for the actual release announcement about Linux/arm64 support.
